aspnetcore/mvc/models/validation.md

```diff
@@ -57,6 +57,18 @@ MVC supports any attribute that derives from `ValidationAttribute` for validatio
 
 There may be instances where you need more features than built-in attributes provide. For those times, you can create custom validation attributes by deriving from `ValidationAttribute` or changing your model to implement `IValidatableObject`.
 
+## Notes on the use of the Required attribute
+
+Non-nullable [value types](/dotnet/csharp/language-reference/keywords/value-types) (such as `decimal`, `int`, `float`, and `DateTime`) are inherently required and don't need the `Required` attribute. The app performs no validation checks server-side for non-nullable types decorated with `Required`.
+
+MVC model binding rejects a property submission in the form data containing an empty string for a non-nullable value type. Model binding usually ignores missing data for properties.
+
+`Required` can be used to change the client-side validation error message for a non-nullable value type property.
```

Comment:
- Up to you but you may want to mention client-side validation always requires a value for these fields.
- Recommend "property" -> "field". "Field" in the HTML sense. An HTML field corresponds to a C# property (usually) but "field" makes more sense here because you're talking about *client-side* validation 😃